bot/events.py

```diff
@@ -0,0 +1,54 @@
+from discord.ext import commands
+from emojis.emojis import emoji_id_mapping, contributors, send_dm_once
+import discord
+import logging
+
+def setup_events(bot: commands.Bot):
+    @bot.event
+    async def on_message(message):
+        for emoji_id, contributor_uid in emoji_id_mapping.items():
+            contributor = next((c for c in contributors if c["uid"] == contributor_uid), None)
```

Comment:
doubly neested loops are tight :)

simple for in would have done the trick just saying

Reply:
Nesting on nesting. I have changed this.